File: SuperiorPackGroup/BLL/InventoryAdjustmentReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class InventoryAdjustmentReportBLL

    Private m_InventoryAdjustmentReportTableAdapter As InventoryAdjustmentReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As InventoryAdjustmentReportTableAdapter
        Get
            If (Me.m_InventoryAdjustmentReportTableAdapter Is Nothing) Then
                Me.m_InventoryAdjustmentReportTableAdapter = New InventoryAdjustmentReportTableAdapter
                Me.m_InventoryAdjustmentReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_InventoryAdjustmentReportTableAdapter
        End Get
    End Property

    Public Function GetInventoryAdjustmentReport(ByVal fromDate As Date?, ByVal toDate As Date?, ByVal customerID As Integer?, _
                                                 ByVal items As String, ByVal inactiveItems As Boolean?, ByVal inactiveCustomers As Boolean?) As SPGReports.InventoryAdjustmentReportDataTable

        If inactiveCustomers.Value = True Then
            inactiveCustomers = Nothing
        End If

        If inactiveItems.Value = True Then
            inactiveItems = Nothing
        End If

        If String.IsNullOrEmpty(items) Then
            Return Adapter.GetInventoryAdjustmentReport(fromDate, toDate, customerID, inactiveCustomers, inactiveItems)
        Else
            Return Adapter.GetInventoryAdjustmentReportByItemID(fromDate, toDate, customerID, inactiveCustomers, items)
        End If

    End Function

End Class

File: SuperiorPackGroup/BLL/InventoryAdjustmentReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class InventoryAdjustmentReportBLL
	{

		private InventoryAdjustmentReportTableAdapter m_InventoryAdjustmentReportTableAdapter = null;

		public InventoryAdjustmentReportTableAdapter Adapter
		{
			get
			{
				if (this.m_InventoryAdjustmentReportTableAdapter == null)
				{
					this.m_InventoryAdjustmentReportTableAdapter = new InventoryAdjustmentReportTableAdapter();
					this.m_InventoryAdjustmentReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return this.m_InventoryAdjustmentReportTableAdapter;
			}
		}

		public SPGReports.InventoryAdjustmentReportDataTable GetInventoryAdjustmentReport(DateTime? fromDate, DateTime? toDate, int? customerID, string items, bool? inactiveItems, bool? inactiveCustomers)
		{

			if (inactiveCustomers.Value == true)
			{
				inactiveCustomers = null;
			}

			if (inactiveItems.Value == true)
			{
				inactiveItems = null;
			}

			if (string.IsNullOrEmpty(items))
			{
				return Adapter.GetInventoryAdjustmentReport(fromDate, toDate, customerID, inactiveCustomers, inactiveItems);
			}
			else
			{
				return Adapter.GetInventoryAdjustmentReportByItemID(fromDate, toDate, customerID, inactiveCustomers, items);
			}

		}

	}

}